front/app/modules/commercial/content_builder/admin/components/ContentBuilderTopBar/index.tsx

```diff
@@ -71,7 +98,12 @@ const ContentBuilderPage = ({ params: { projectId } }) => {
             </>
           )}
         </Box>
-        <Button buttonStyle="primary" onClick={goBack}>
+        <Button
+          buttonStyle="primary"
+          processing={loading}
+          onClick={handleSave}
+          data-testid="contentBuilderTopBarSaveButton"
+        >
```

Comment:
Mostly a question for my own knowledge - in which case is the data-testid preferred? In some cases it has been recommended to me to "getAllByRole" and then select the object I need from that. And when I'm reading online people usually recommend using data-testid as a last resort, though I'm not entirely certain why?

Reply:
Normally it is not preferred - the reason is that RTL tries to mimic what the user sees/ can interact with and they don't see data test ids. So normally, you would get a button by role but relying on the index of the element is also not great because it's an implementation detail in a way. Targeting the text is usually the way to go but the problem there is that the copy is controlled by product outside of the codebase and if we use that, it might change and the test will fail. 

So while it's not ideal - we use data test id in this case because it's safest.